web_app/models/AC_model_benarie1986.py

```diff
@@ -60,6 +62,16 @@ def _display_site_info(self, corrosion_site: int) -> None:
             """
         )
 
+    def _show_map(self, parameters) -> None:
+         # Show the selected location on a map
+        coordinates = pd.read_csv(self.COORDINATES_FILE_PATH, header=None)
+        coordinates = coordinates.iloc[parameters['corrosion_site'], 1:]
+        coordiantes = pd.DataFrame({
+            'lat': [float(coordinates.iloc[0])],
+            'lon': [float(coordinates.iloc[1])]
+        })
+        st.map(coordiantes)
+
```

Comment:
This function shouldn't be in the model.
1. Load the coordinates in the model.
2. If coordinates are available for a model show them in the model view

Reply:
check the new commit